### PR TITLE
W3C traceparent

### PR DESCRIPTION
**Pull request by cleptric**
https://github.com/getsentry/team-sdks/issues/41

### CODE REVIEW

File: src/docs/sdk/performance/dynamic-sampling-context.mdx [hunk 1]
```diff
@@ -197,7 +197,7 @@ def has_sentry_value_in_baggage_header(request):
   # header of `request`, for which the key starts with "sentry-". Otherwise, it returns False.
 
 def on_incoming_request(request):
-  if request.has_header("sentry-trace") and (not request.has_header("baggage") or not has_sentry_value_in_baggage_header(request)):
+  if request.has_header("sentry-trace") or request.has_header("traceparent") and (not request.has_header("baggage") or not has_sentry_value_in_baggage_header(request)):
```
Comment (stephanie-anderson) [inline src/docs/sdk/performance/dynamic-sampling-context.mdx:200]: `if (request.has_header("sentry-trace") or request.has_header("traceparent")) and (not request.has_header("baggage") or not has_sentry_value_in_baggage_header(request)):`

File: src/docs/sdk/performance/dynamic-sampling-context.mdx [hunk 2]
```diff
@@ -64,7 +64,7 @@ Baggage is a standard HTTP header with URI encoded key-value pairs.
 For the propagation of DSC, SDKs first read the DSC from the baggage header of incoming requests/messages.
 To propagate DSC to downstream SDKs/services, we create a baggage header (or modify an existing one) through HTTP request instrumentation.
 
-The `baggage` header should only be attached to an outgoing request if the request's URL matches at least one entry of the [`tracePropagationTargets`](/sdk/performance/#tracepropagationtargets) SDK option.
+The `baggage` header should only be attached to an outgoing request if the request's URL matches at least one entry of the [`tracePropagationTargets`](/sdk/performance/#tracepropagationtargets) SDK option or this option is set to `null`.
```
Comment (Lms24) [inline src/docs/sdk/performance/dynamic-sampling-context.mdx:67]: l: 
```suggestion
The `baggage` header should only be attached to an outgoing request if the request's URL matches at least one entry of the [`tracePropagationTargets`](/sdk/performance/#tracepropagationtargets) SDK option or this option is set to `null` or not set.
```

**Review by mydea — APPROVED** [review]

File: src/docs/sdk/performance/index.mdx [hunk 3]
```diff
@@ -117,7 +117,8 @@ tree as well as the unit of reporting to Sentry.
   - `Span` should have a method `startChild` which creates a new span with the current span's id as the new span's `parentSpanId` and the current span's `sampled` value copied over to the new span's `sampled` property
   - The `startChild` method should respect the `maxSpans` limit, and once the limit is reached the SDK should not create new child spans for the given transaction.
   - `Span` should have a method called `toSentryTrace` which returns a string that could be sent as a header called `sentry-trace`.
-  - `Span` should have a method called `iterHeaders` (adapt to platform's naming conventions) that returns an iterable or map of header names and values. This is a thin wrapper containing `return {"sentry-trace": toSentryTrace()}` right now. See `continueFromHeaders` as to why this exists and should be preferred when writing integrations.
+  - `Span` should have a method called `toW3CTrace` which returns a string that could be sent as a header called `traceparent`.
```
Comment (Lms24) [inline src/docs/sdk/performance/index.mdx:120]: side note: These docs here assume our SDKs create spans themselves. This is no longer the case in Otel-based SDKs (e.g. JS Node), so we can't define most of these methods on a span/transaction interface. Instead we expose top level helper functions. I think it's fine to keep things as-is in this PR but we probably wanna rework this page a bit to keep things up to date.

File: src/docs/sdk/performance/index.mdx [hunk 4]
```diff
@@ -220,6 +222,16 @@ Besides the [usual reasons to use \*defer](https://github.com/apache/incubator-z
...
 
+## Header `traceparent`
+
+The header is used for trace propagation. SDKs use the header to continue traces from upstream services (e.g. incoming HTTP requests), and to propagate tracing information to downstream services (e.g. outgoing HTTP requests).
+
+`traceparent = version-traceid-spanid-traceflags`
+
+See [W3C `traceparent` header](https://www.w3.org/TR/trace-context/#traceparent-header) for more information.
+
+The `traceparent` header should only be attached to an outgoing request if the request's URL matches at least one entry of the [`tracePropagationTargets`](#tracepropagationtargets) SDK option or this option is set to `null`.
```
Comment (Lms24) [inline src/docs/sdk/performance/index.mdx:233]: l: (for consistency but feel free to ignore)
```suggestion
The `traceparent` header should only be attached to an outgoing request if the request's URL matches at least one entry of the [`tracePropagationTargets`](#tracepropagationtargets) SDK option or this option is set to `null` or not set.
```

File: src/docs/sdk/performance/index.mdx [hunk 5]
```diff
@@ -220,6 +222,16 @@ Besides the [usual reasons to use \*defer](https://github.com/apache/incubator-z
 
 Which in reality is useful for proxies to set it to `0` and opt out of tracing.
 
+## Header `traceparent`
+
+The header is used for trace propagation. SDKs use the header to continue traces from upstream services (e.g. incoming HTTP requests), and to propagate tracing information to downstream services (e.g. outgoing HTTP requests).
+
+`traceparent = version-traceid-spanid-traceflags`
```
Comment (Lms24) [inline src/docs/sdk/performance/index.mdx:229]: Should we explain `version` and `traceflags` a bit more? 
From our earlier conversation and skimming over the spec, `version` should be set to `00` and `traceflags` to `00`/`01` - correct?

Comment (cleptric) [reply src/docs/sdk/performance/index.mdx:229]: Let me link to https://www.w3.org/TR/trace-context/#traceparent-header here as well.

**Review by Lms24 — APPROVED** [review]

**Pull request merged**